web/client/components/mapcontrols/searchcoordinates/CoordinatesSearch.js

```diff
@@ -131,9 +163,30 @@ const CoordinatesSearch = ({
 
     const changeCoordinates = (coord, value) => {
         onChangeCoord(coord, parseFloat(value));
+        // set current map crs to coordinate object
+        if (coordinate?.currentMapXYCRS !== currentMapCRS && currentMapCRS !== "EPSG:4326") onChangeCoord('currentMapXYCRS', currentMapCRS);
         if (!areValidCoordinates()) {
             onClearCoordinatesSearch({owner: "search"});
         }
+        // if there is mapCRS available --> calculate X/Y values by reproject to display in case switch to MapCRS
```

Comment:
This seems to be missing in
https://github.com/geosolutions-it/MapStore2/pull/10220/files#diff-455b07539c76115f6dce6c982a88461c05ae2ab7cb8a2a3fac5a2a6ad58ba9cc